File: src/main/java/graphicscore/placeholder/PlaceholderExtension.java
package graphicscore.placeholder;

import cn.nukkit.Player;
import cn.nukkit.event.Listener;
import cn.nukkit.plugin.Plugin;

public abstract class PlaceholderExtension implements Listener {
    public abstract Plugin getOwnedPlugin();

    public abstract String getIdentifier();

    public abstract String onRequest(Player player, String params);
}

File: src/main/kotlin/me/skh6075/pnx/graphicscore/placeholder/PlaceholderExtension.kt
package me.skh6075.pnx.graphicscore.placeholder

import cn.nukkit.Player
import cn.nukkit.event.Listener
import cn.nukkit.plugin.Plugin

abstract class PlaceholderExtension : Listener {
    abstract fun getOwnedPlugin(): Plugin

    abstract fun getIdentifier(): String

    abstract fun onRequest(player: Player, params: String) : String?
}